ui-frontend/packages/catalog-ui-search/src/main/webapp/js/model/User.js

```diff
@@ -499,11 +499,29 @@ User.Response = Backbone.AssociatedModel.extend({
   canRead(metacard) {
     return new Security(Restrictions.from(metacard)).canRead(this)
   },
-  canWrite(metacard) {
-    return (
-      new Security(Restrictions.from(metacard)).canWrite(this) &&
-      metacard.get('createdBy') !== 'system'
-    )
+  /**
+   * Allow workspaces to be created no matter what,
+   * but still enforce the other restrictions (if it's a shared workspace,
+   * check if they have permission to write to it)
+   *
+   * For all other types, every policy is enforced.
+   */
+  canWrite(thing) {
```

Comment:
Looks like this `isEditingRestricted` is set by a backend property `isEditingAllowed`. I think for the result forms purpose, we will leverage this property and it should correctly return false here. However, for the converse case (when the result form is editable), the logic is broken for result forms. They currently don't have a `type` property, so the `default` case below will be hit, causing a console error.

It seems like the `default` case below is assuming that there will be a collection of some sort if it is hit, is that correct?

Reply:
the added check for `undefined` in the `default` case below handles the result forms, so no addiitonal changes should be needed for them. thanks!

Reply:
@cassandrabailey293 - made some updates to this after speaking with @andrewkfiedler .  This logic now appears to work as expected in DDF.  It would need verification for downstream projects though.  Note that I removed the editingEnabled property as I don't think it is needed since this change determines editability based on the metacard's attributes.